### ISSUE TITLE
IBeacon plugin not working using Ionic v4 beta

### ISSUE THREAD

**Issue by elmergu** [ionic-team/ionic-native#2631]
**I'm submitting a ...**
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or https://ionicworldwide.herokuapp.com/

**Current behavior:**
the following error is displayed in xcode when running from the device

_ERROR: [Cordova-Plugin-IBeacon] WARNING did not receive delegate ready callback from dom after 30 seconds_

**Expected behavior:**
beacon ranging and monitoring should start

**Steps to reproduce:**

_Using the start app_
ionic start myApp tabs --type=angular

_Installing the ibeacon plugin_
ionic cordova plugin add cordova-plugin-ibeacon
$ npm install --save @ionic-native/ibeacon@beta

_adding in the app.module.ts the plugin_
```
import { IBeacon } from '@ionic-native/ibeacon/ngx';

@NgModule({
  declarations: [AppComponent],
  entryComponents: [],
  imports: [BrowserModule, IonicModule.forRoot(), AppRoutingModule],
  providers: [
    StatusBar,
    SplashScreen,
    IBeacon,
    { provide: RouteReuseStrategy, useClass: IonicRouteStrategy }
  ],
  bootstrap: [AppComponent]
})
export class AppModule {}
```

**Related code:**

home.page.ts
```
import { Component } from '@angular/core';
import { IBeacon, IBeaconPluginResult } from '@ionic-native/ibeacon/ngx';
import { Platform } from '@ionic/angular';

@Component({
  selector: 'app-home',
  templateUrl: 'home.page.html',
  styleUrls: ['home.page.scss']
})
export class HomePage {
  private isAdvertisingAvailable: boolean = null;

  private uuid = '11111111-1111-1111-1111-111111111111';
  beaconData = {};
  beaconUuid: String;

  constructor(private readonly ibeacon: IBeacon, private readonly platform: Platform) {
    this.enableDebugLogs();
  }

  public enableDebugLogs(): void {
    this.platform.ready().then(async () => {
      this.ibeacon.enableDebugLogs();
      this.ibeacon.enableDebugNotifications();
    });
  }

  public onStartClicked(): void {
    this.platform.ready().then(() => {
      this.startBleFun();
    });
  }

  public startBleFun(): void {

    // Request permission to use location on iOS
    this.ibeacon.requestAlwaysAuthorization();
    // create a new delegate and register it with the native layer
    const delegate = this.ibeacon.Delegate();

    this.ibeacon.setDelegate(delegate);
    this.beaconUuid = this.uuid;
    // Subscribe to some of the delegate's event handlers
    delegate.didRangeBeaconsInRegion().subscribe(
      (pluginResult: IBeaconPluginResult) => console.log('didRangeBeaconsInRegion: ', pluginResult),
      (error: any) => console.error(`Failure during ranging: `, error)
    );
    delegate.didStartMonitoringForRegion().subscribe(
      (pluginResult: IBeaconPluginResult) => console.log('didStartMonitoringForRegion: ', pluginResult),
      (error: any) => console.error(`Failure during starting of monitoring: `, error)
    );

    delegate.didEnterRegion().subscribe(
      (pluginResult: IBeaconPluginResult) => {
        console.log('didEnterRegion: ', pluginResult);
        this.beaconData = pluginResult;
      }
    );

    delegate.didExitRegion().subscribe(
      (pluginResult: IBeaconPluginResult) => {
        console.log('didExitRegion: ', pluginResult);
      }
    );

    console.log(`Creating BeaconRegion with UUID of: `, this.uuid);
    const beaconRegion = this.ibeacon.BeaconRegion('nullBeaconRegion', this.uuid, 1, 1);

    this.ibeacon.startMonitoringForRegion(beaconRegion).then(
      () => console.log('Native layer recieved the request to monitoring'),
      (error: any) => console.error('Native layer failed to begin monitoring: ', error)
    );

    this.ibeacon.startRangingBeaconsInRegion(beaconRegion)
      .then(() => {
        console.log(`Started ranging beacon region: `, beaconRegion);
      })
      .catch((error: any) => {
        console.error(`Failed to start ranging beacon region: `, beaconRegion);
      });
  }

}
```

home.page.html
```
<ion-header>
  <ion-toolbar>
    <ion-title>Home</ion-title>
  </ion-toolbar>
</ion-header>

<ion-content padding>
    <h3>Cordova Proximity Quickstart</h3>
  <ion-list>

    <ion-item>
      <ion-label fixed>UUID</ion-label>
      <ion-textarea type="text" [(ngModel)]="uuid"></ion-textarea>
    </ion-item>

  </ion-list>

  <div>
    <div>beacon DATA</div>
    <div>{{beaconUuid}}</div>
    {{beaconData | json}}
  </div>

  <ion-button shape="round" (click)="onStartClicked()" color="primary" fill="outline">Start</ion-button>

  
</ion-content>
```

**Other information:**

**Ionic info:**

```
Ionic:

   ionic (Ionic CLI)          : 4.0.3 (/Users/estuardo/.npm-packages/lib/node_modules/ionic)
   Ionic Framework            : @ionic/angular 4.0.0-beta.0
   @angular-devkit/core       : 0.7.0-rc.3
   @angular-devkit/schematics : 0.7.0-rc.3
   @angular/cli               : 6.0.8
   @ionic/ng-toolkit          : 1.0.0
   @ionic/schematics-angular  : 1.0.1

Cordova:

   cordova (Cordova CLI) : 8.0.0
   Cordova Platforms     : ios 4.5.5

System:

   NodeJS : v8.11.3 (/usr/local/bin/node)
   npm    : 5.8.0
   OS     : macOS High Sierra
   Xcode  : Xcode 9.4.1 Build version 9F2000

Environment:

   ANDROID_HOME : not set
```

**Comment by fengyanliang** [comment]
I have the same issue, have you fixed it now?

**Comment by halindrome** [comment]
I also have the same issue.

**Comment by hhetland** [comment]
I also have problems with the iBeacon plugin, using 5.0.0.0-beta.20 but same error on previous versions.
On IOS this shows up in the developer console in Safari, any clues there?:

Error: Uncaught (in promise): TypeError: undefined is not an object (evaluating 'new _ionic_native_core__WEBPACK_IMPORTED_MODULE_1__["cordova"].plugins.locationManager')

**Comment by danielsogl** [comment]
cc @mlynch can you take a look for me?

**Comment by hhetland** [comment]
No changes in beta 21, same error as described above

**Comment by raujonas** [comment]
Does anyone know if there's something new about this issue? I have the same issue with the location manager

**Comment by matteocusinato** [comment]
I have the same issue, please solve it!!

**Comment by matteobortolazzo** [comment]
Please fix it.

**Comment by kurtommy** [comment]
Hi @danielsogl @mlynch  here you have an example of not working plugin
https://github.com/kurtommy/ionic-ibeacon-broken

i've created new app from scratch with the CLI (ionic version 4)
then followed steps found here
https://beta.ionicframework.com/docs/native/ibeacon

but i have in console
ERROR Error: Uncaught (in promise): TypeError: Cannot read property 'locationManager' of undefined

**Comment by gischy** [comment]
same issue

**Comment by kurtommy** [comment]
@ihadeed @danielsogl Hi guys some help on this? we are building a nice app to track children that are taking the bus to send notifications to parents if they have took the bus or not. we are using ionic and beacon but this plugin is not working...

**Comment by MiguelSyll** [comment]
I have the same problem but with the new realase of V4. Any solution or idea to solve this problem?

**Comment by EdByrnee** [comment]
Any solution yet guys?

**Comment by bitforcesrl** [comment]
nothing here.. i've asked same question on the ionic slack to Mike an ionic mantainer he asked me for a live broken example but i don't have time now to perpare it, someone has some basic example to share with him?

https://ionic-worldwide.slack.com/archives/C03AJV5A5/p1548779396066400

**Comment by matteocusinato** [comment]
> nothing here.. i've asked same question on the ionic slack to Mike an ionic mantainer he asked me for a live broken example but i don't have time now to perpare it, someone has some basic example to share with him?
> 
> https://ionic-worldwide.slack.com/archives/C03AJV5A5/p1548779396066400

Ok, done.

**Comment by EdByrnee** [comment]
Thank you

> > nothing here.. i've asked same question on the ionic slack to Mike an ionic mantainer he asked me for a live broken example but i don't have time now to perpare it, someone has some basic example to share with him?
> > https://ionic-worldwide.slack.com/archives/C03AJV5A5/p1548779396066400
> 
> Ok, done.

Thank you, we all appreciate it. Hopefully we hear back soon.

For anyone looking for an interim solution, I've managed to get the plugin working in Ionic v3 in the foreground and background but when the app is closed it is not relaunching the application. If anyone has any solution it would be greatly appreciated. Thanks

**Comment by arneldomingo** [comment]
I got it to work. A hack but it worked. I forced cordova.plugins.location.locationManager in IBeacon.prototype.Delegate of ../ibeacon/ngx/index.js to take the value of pluginInstance in the function checkAvailability of ../core/decorators/common.js.

**Comment by matteocusinato** [comment]
@arneldomingo can you tell me a little more? Thanks

**Comment by arneldomingo** [comment]
_* All paths I refer to in this reply has root node_modules/@ionic_native_

In ./ibeacon/ngx/index.js, there are 2 places that are breaking:
1) line 32 inside function IBeacon.prototype.Delegate 
`var delegate = new cordova.plugins.locationManager.Delegate();`
and, 
2) line 96 inside function IBeacon.prototype.BeaconRegion
`return new cordova.plugins.locationManager.BeaconRegion(identifer, uuid, major, minor, notifyEntryStateOnDisplay);`

Everybody knows that the culprit is  cordova.plugins.locationManager being undefined. 

Now, in ./core/decorators/common.js, there is function **checkAvailability**  starting at line 132.
In line 142, the variable **pluginInstance** is populated 
`pluginInstance = getPlugin(pluginRef);`

**pluginInstance** SHOULD BE THE VALUE OF cordova.plugins.locationManager. You'll need to figure out a way to get that and replace cordova.plugins.locationManager with that value.

For me as a QUICK hack (you might have a more elegant solution), I copied the function checkAvailability (rename it to checkAvailability2, for example) , but return pluginInstance instead of returning true. 
```
export function checkAvailability2(plugin, methodName, pluginName) {
	.
	.
	.
  return pluginInstance;
}
```

SO back to  ./ibeacon/ngx/index.js,
I first import the new function in line 21:

`import { IonicNativePlugin, cordova, checkAvailability, checkAvailablity2 } from '@ionic-native/core';`

Then I declare a class level variable:
`var locationManager;`

Then, I inserted the line, just before line 32:
` _this.locationManager = checkAvailability2(_this);`
And replace line 32 with
`var delegate = new  _this.locationManager.Delegate(); `

I used the same variable to replace line 96 with:
`return new _this.locationManager.BeaconRegion(identifer, uuid, major, minor, notifyEntryStateOnDisplay);`

Again this is a hack and works for me. I need a quick solution since I have a deadline for a prototype project. But this is an idea to fix this. Somebody needs to come out with an official solution.

Hope this helps.

**Comment by Linkmetal** [comment]
@arneldomingo It worked for me! Literally saved my life.

**Comment by EdByrnee** [comment]
Thank you! Got this working and monitoring iBeacons in the background and foreground. I'll make a PR

 (Still no luck with monitoring when the app is closed)

**Comment by Linkmetal** [comment]
Idk if it's intended because is my first time working with beacons,when I detect a beacon with a certain uuid, I can't detect it anymore for some minutes, even restarting the app. Maybe is due to the files that we modified to fix the bug, or is there any kind of timer to detect a beacon again?

**Comment by MiguelSyll** [comment]
Hey @arneldomingo.
Do you have a copy of the file /ibeacon/ngx/index.js?

It could be good to compare with my file index.js

Thanks!

**Comment by tommasorossi** [comment]
This WE i found time to try all my beacons, debugging the plugin it looks like that cordova variable is shadowed
In ./ibeacon/ngx/index.js,
i've replaced
`cordova.plugins.locationManager.Delegate();`
with 
`window.cordova.plugins.locationManager.Delegate();`

but there is no point to use this lib which is not mantained anymore.
Instead there is this great plugin which works in just one line of code with Eddystone (which is more capable than iBeacon from Apple)
It works perfectly

https://www.npmjs.com/package/cordova-plugin-eddystone

**Comment by elTigre9** [comment]
@tommasorossi 
That did the trick for me. 

I created a plugin for ibeacon in Swift that's largely unfinished. It connects and ranges, but the delegate callbacks don't work, so you'd have to go into the Swift code and customize it there. I'm going to keep working on it, just in case people don't want to toss their iBeacons in the trash.

**Comment by giladrom** [comment]
@tommasorossi iOS will not wake up in the background for Eddystone beacons - which are also not supported anymore by Google.

**Comment by elTigre9** [comment]
Hi everyone. I have the new beacon plugin setup on my account: https://github.com/elTigre9/cordova-plugin-beaconing. I hope it comes in handy!

**NOTES:**

1. Currently, it's only for iOS. 
2. You can only run 1 beacon at a time
3. You can only range and monitor (enter and exit beacon region)
4. It's written in Swift 4
5. Still a work in progress, but I feel like it's got potential

**Comment by IT-Dan** [comment]
Here's a copy of those files that need to be modified https://github.com/IT-Dan/Ionic4ibeacon/.

**Comment by EdByrnee** [comment]
I cannot find any working solution for monitoring iBeacons when the app is closed. This repo is the closest I can get to it, it works in the foreground/background but not when the app is closed. Does anyone know of a working solution or will I have to learn swift?

**Comment by RangerRick** [comment]
I can confirm that the patch to use `cordova.plugins` rather than `window.cordova.plugins` works for me as well. I checked out https://github.com/IT-Dan/Ionic4ibeacon/ and changed my import from `@ionic-native/ibeacon/ngx` to `3rdparty/Ionic4ibeacon/ngx` and I'm getting ibeacon events.

**Comment by cagb80** [comment]
> @arneldomingo It worked for me! Literally saved my life.
What was the @arneldomingo solution?

**Comment by cagb80** [comment]
> I can confirm that the patch to use `cordova.plugins` rather than `window.cordova.plugins` works for me as well. I checked out https://github.com/IT-Dan/Ionic4ibeacon/ and changed my import from `@ionic-native/ibeacon/ngx` to `3rdparty/Ionic4ibeacon/ngx` and I'm getting ibeacon events.

Could you tell me where do you put the files :)

**Comment by cagb80** [comment]
> Here's a copy of those files that need to be modified https://github.com/IT-Dan/Ionic4ibeacon/.

This worked for me... @IT-Dan TY

**Comment by bramji06** [comment]
Still not working ionic 4. Can you help 
"
vendor-es2015.js:46777 ERROR Error: Uncaught (in promise): TypeError: Cannot read property 'locationManager' of undefined
TypeError: Cannot read property 'locationManager' of undefined
    at vendor-es2015.js:126034
    at IBeacon.push../node_modules/@ionic-native/ibeacon/ngx/index.js.IBeacon.Delegate (vendor-es2015.js:126092)
"
Already implement suggestion as below

i've replaced
cordova.plugins.locationManager.Delegate();
with
window.cordova.plugins.locationManager.Delegate();

**Comment by kodeine** [comment]
> This WE i found time to try all my beacons, debugging the plugin it looks like that cordova variable is shadowed
> In ./ibeacon/ngx/index.js,
> i've replaced
> `cordova.plugins.locationManager.Delegate();`
> with
> `window.cordova.plugins.locationManager.Delegate();`
> 
> but there is no point to use this lib which is not mantained anymore.
> Instead there is this great plugin which works in just one line of code with Eddystone (which is more capable than iBeacon from Apple)
> It works perfectly
> 
> https://www.npmjs.com/package/cordova-plugin-eddystone

can you share how you used that plugin in ionic?

**Comment by atmosuwiryo** [comment]
It is said that this is ionic-native bug here, dated 28-03-2019.
https://github.com/ionic-team/capacitor/issues/1367#issuecomment-477659127
Still no bug fix until now?

**Comment by kodeine** [comment]
anyone was able to make advertising work? to me  `this.ibeacon.isAdvertising()` coming as false even though i started advertising.

**Comment by RangerRick** [comment]
FYI, if you're trying to do beacon stuff under Capacitor, I've written a Capacitor plugin for it: https://github.com/rangerrick/capacitor-ibeacon

Haven't had a chance to implement the Android side yet, so it's iOS-only for now.

**Comment by kodeine** [comment]
i'm looking for both, but any idea's why the advertising doesnt work for IBeacon? am i missing something?


home.ts
```
import {Component} from '@angular/core';
import {IonicPage, Platform} from 'ionic-angular';
import {IBeacon, IBeaconPluginResult} from "@ionic-native/ibeacon/ngx";

@IonicPage()
@Component({
  selector: 'ble-home',
  templateUrl: 'ble_home.html'
})
export class BleHomePage {
  public bleData: any = {};
  public uuid: string = '00000000-0000-0000-0000-000000000000';
  // public uuid: string = '11111111-1111-1111-1111-111111111111';
  public major: number = 11;
  public minor: number = 11;
  public identifier: string = 'deskBeacon';

  constructor(private readonly ibeacon: IBeacon, private readonly platform: Platform) {
    this.enableDebugLogs();
  }

  public enableDebugLogs(): void {
    this.platform.ready().then(async () => {
      this.ibeacon.enableDebugLogs();
      this.ibeacon.enableDebugNotifications();
    });
  }

  public onStartClicked(): void {
    this.platform.ready().then(() => {
      this.startBleFun();
    });
  }

  public startBleFun(): void {

    // Request permission to use location on iOS
    this.ibeacon.requestAlwaysAuthorization();

    // create a new delegate and register it with the native layer
    let delegate = this.ibeacon.Delegate();

    // Subscribe to some of the delegate's event handlers
    delegate.didRangeBeaconsInRegion().subscribe(
      (pluginResult: IBeaconPluginResult) => console.log('didRangeBeaconsInRegion: ', pluginResult),
      (error: any) => console.error(`Failure during ranging: `, error)
    );
    delegate.didStartMonitoringForRegion().subscribe(
      (pluginResult: IBeaconPluginResult) => console.log('didStartMonitoringForRegion: ', pluginResult),
      (error: any) => console.error(`Failure during starting of monitoring: `, error)
    );

    delegate.didEnterRegion().subscribe(
      (pluginResult: IBeaconPluginResult) => {
        this.bleData = pluginResult;
        console.log('didEnterRegion: ', pluginResult);
      }
    );

    delegate.didExitRegion().subscribe(
      (pluginResult: IBeaconPluginResult) => {
        console.log('didExitRegion: ', pluginResult);
      }
    );

    console.log(`Creating BeaconRegion with UUID of: `, this.uuid);
    const beaconRegion = this.ibeacon.BeaconRegion(this.identifier, this.uuid);

    this.ibeacon.startMonitoringForRegion(beaconRegion).then(
      () => console.log('Native layer recieved the request to monitoring'),
      (error: any) => console.error('Native layer failed to begin monitoring: ', error)
    );

    this.ibeacon.startRangingBeaconsInRegion(beaconRegion)
      .then(() => {
        console.log(`Started ranging beacon region: `, beaconRegion);
      })
      .catch((error: any) => {
        console.error(`Failed to start ranging beacon region: `, beaconRegion);
      });
  }

}
```

advertising
```
import {Component} from '@angular/core';
import {IonicPage, Platform} from 'ionic-angular';
import {IBeacon, IBeaconPluginResult} from "@ionic-native/ibeacon/ngx";
import { interval } from 'rxjs/observable/interval';
import {Subject} from "rxjs/Subject";
import {takeUntil} from "rxjs/operators";

@IonicPage()
@Component({
  selector: 'page-advertising',
  templateUrl: 'ble_advertise.html'
})
export class BleAdvertisePage {

  destroy$: Subject<boolean> = new Subject<boolean>();
  private settings: any = {};

  public uuid: string = '11111111-1111-1111-1111-111111111111';
  public major: number = 11;
  public minor: number = 11;
  public identifier: string = 'deskBeacon';

  constructor(private readonly ibeacon: IBeacon, private readonly platform: Platform) {

  }

  public enableDebugLogs(): void {
    this.platform.ready().then(async () => {
      this.ibeacon.enableDebugLogs();
      this.ibeacon.enableDebugNotifications();
    });
  }

  public fetchIsAdvertisingAvailable(): void {
    this.platform.ready().then(async () => {
      this.settings = {
        'isAdvertising': await this.ibeacon.isAdvertising(),
          'isBluetoothEnabled': await this.ibeacon.isBluetoothEnabled(),
          'isAdvertisingAvailable': await this.ibeacon.isAdvertisingAvailable(),
      };
    });
  }

  public onAdvertiseClicked(): void {
    this.platform.ready().then(() => {
      this.enableDebugLogs();
      this.startBleAdvertising();

      interval(5*1000).pipe(takeUntil(this.destroy$)).subscribe(() => {
        this.fetchIsAdvertisingAvailable();
      });
    });
  }

  public onStopAdvertiseClicked(): void {
    this.platform.ready().then(() => {
      const beaconRegion = this.ibeacon.BeaconRegion(this.identifier, this.uuid);
      this.ibeacon.stopAdvertising(beaconRegion)
        .then((e) => {
          console.log(`AdvertisingPage::onStopAdvertiseClicked::SUCCESS=>`, e);

          if (this.destroy$) {
            console.log('Interval stopped.');
            this.destroy$.next(true);
            // this.destroy$.unsubscribe();
        }
        })
        .catch((reason: any) => {
          console.log(`AdvertisingPage::onStopAdvertiseClicked::FAIL::reason=>`, reason);
        });
    });
  }

  public startBleAdvertising(): void {

    // Request permission to use location on iOS
    this.ibeacon.requestAlwaysAuthorization();
    this.ibeacon.requestWhenInUseAuthorization();

    this.ibeacon.enableDebugLogs();

    // create a new delegate and register it with the native layer
    let delegate = this.ibeacon.Delegate();

    // Event when advertising starts (there may be a short delay after the request)
    // The property 'region' provides details of the broadcasting Beacon
    delegate.peripheralManagerDidStartAdvertising().subscribe((pluginResult: IBeaconPluginResult) => {
      console.log('peripheralManagerDidStartAdvertising', pluginResult);
    }, err => {
      console.error('peripheralManagerDidStartAdvertising', err);
    });

    // Event when bluetooth transmission state changes
    // If 'state' is not set to BluetoothManagerStatePoweredOn when advertising cannot start
    delegate.peripheralManagerDidUpdateState().subscribe((pluginResult: IBeaconPluginResult) => {
      console.log('peripheralManagerDidUpdateState', pluginResult);
    }, err => {
      console.error('peripheralManagerDidUpdateState', err);
    });

    const beaconRegion = this.ibeacon.BeaconRegion(this.identifier, this.uuid, this.major, this.minor);
    this.ibeacon.startAdvertising(beaconRegion).then((e) => console.log('Advertising started', e));
  }

}

```

**Comment by mohit810** [comment]
@EdByrnee can u provide ur working code , please.?

**Comment by mohit810** [comment]
> I cannot find any working solution for monitoring iBeacons when the app is closed. This repo is the closest I can get to it, it works in the foreground/background but not when the app is closed. Does anyone know of a working solution or will I have to learn swift?

@EdByrnee  can u provide ur working code , please.?

**Comment by mohit810** [comment]
> This WE i found time to try all my beacons, debugging the plugin it looks like that cordova variable is shadowed
> In ./ibeacon/ngx/index.js,
> i've replaced
> `cordova.plugins.locationManager.Delegate();`
> with
> `window.cordova.plugins.locationManager.Delegate();`
> 
> but there is no point to use this lib which is not mantained anymore.
> Instead there is this great plugin which works in just one line of code with Eddystone (which is more capable than iBeacon from Apple)
> It works perfectly
> 
> https://www.npmjs.com/package/cordova-plugin-eddystone

@tommasorossi i am not able to use that lib. as when i try to add it as a plugin only scanning code gets add as plugin not the advertising code.

**Comment by MiguelAngelBrito** [comment]
> > This WE i found time to try all my beacons, debugging the plugin it looks like that cordova variable is shadowed
> > In ./ibeacon/ngx/index.js,
> > i've replaced
> > `cordova.plugins.locationManager.Delegate();`
> > with
> > `window.cordova.plugins.locationManager.Delegate();`
> > but there is no point to use this lib which is not mantained anymore.
> > Instead there is this great plugin which works in just one line of code with Eddystone (which is more capable than iBeacon from Apple)
> > It works perfectly
> > https://www.npmjs.com/package/cordova-plugin-eddystone
> 
> @tommasorossi i am not able to use that lib. as when i try to add it as a plugin only scanning code gets add as plugin not the advertising code.



> I also have a _similar_ issue. In my case, debugging shows that the android app is throwing an error when attempting to create the Delegate object:
> 
> `let d = this.ibeacon.Delegate();`
> 
> This throws an error in the ionic-native ngx module from the line that tries to create a new Delegate calling the cordova plugin (line 29 of ngx/index.js):
> 
> `var delegate = new cordova.plugins.locationManager.Delegate();`
> 
> The error I see in the Chrome debugger console is:
> 
> `ERROR Error: Uncaught (in promise): TypeError: Cannot read property 'locationManager' of undefined`
> 
> Single stepping through it, it seems like the cordova.plugins interface is set up, so I don't really know what it is complaining about. Thoughts?

I had the same issue,   the origin of problem is the version android try with it
ionic cordova platform add android@^8.1.0-nightly.2019.9.10.42c0cba7

**Comment by MiguelAngelBrito** [comment]
> I also have a _similar_ issue. In my case, debugging shows that the android app is throwing an error when attempting to create the Delegate object:
> 
> `let d = this.ibeacon.Delegate();`
> 
> This throws an error in the ionic-native ngx module from the line that tries to create a new Delegate calling the cordova plugin (line 29 of ngx/index.js):
> 
> `var delegate = new cordova.plugins.locationManager.Delegate();`
> 
> The error I see in the Chrome debugger console is:
> 
> `ERROR Error: Uncaught (in promise): TypeError: Cannot read property 'locationManager' of undefined`
> 
> Single stepping through it, it seems like the cordova.plugins interface is set up, so I don't really know what it is complaining about. Thoughts?

I had the same issue, the origin of problem is the version android try with it
ionic cordova platform add android@^8.1.0-nightly.2019.9.10.42c0cba7

**Comment by YouQam** [comment]
> Here's a copy of those files that need to be modified https://github.com/IT-Dan/Ionic4ibeacon/.

Thanks! 
It used to work with me in ionic4 but I realized that in ionic5 the file `ivy_ngcc/ngx/index.js` should be modified as well.

You can find the files to be modified in ionic5 in  https://github.com/YouQam/ionic5ibeacon

The solution has been tested using both iOS and Android devices.

**Comment by Stefanic** [comment]
I made a pull request for the Ionic Native wrapper of this library last week and it has already been released with version 5.28.0:
https://github.com/ionic-team/ionic-native/pull/3505

Hope it helps you guys!
